distributed/worker.py

```diff
@@ -3505,12 +3489,15 @@ def max2(workers: set[str]) -> list[str]:
 
             ts.who_has = workers
             # currently fetching -> can no longer be fetched -> transition to missing
+            # currently missing -> opportunity to be fetched -> transition to fetch
             # any other state -> eventually, possibly, the task may transition to fetch
             # or missing, at which point the relevant transitions will test who_has that
             # we just updated. e.g. see the various transitions to fetch, which
             # instead recommend transitioning to missing if who_has is empty.
             if not workers and ts.state == "fetch":
                 recs[ts] = "missing"
+            elif workers and ts.state == "missing":
+                recs[ts] = "fetch"
```

Comment:
Note that this could be triggered by any of the 4 events that invoke update_who_has:
- handle_compute_task
- handle_acquire_replicas
- RefreshWhoHasEvent, triggered by find_missing
- RefreshWhoHasEvent, triggered by refresh_who_has